winpr/libwinpr/file/generic.c

```diff
@@ -890,7 +890,7 @@ static BOOL ConvertFindDataAToW(LPWIN32_FIND_DATAA lpFindFileDataA,
 	if (length > MAX_PATH)
 		length = MAX_PATH;
 
-	CopyMemory(lpFindFileDataW->cFileName, unicodeFileName, length);
+	CopyMemory(lpFindFileDataW->cFileName, unicodeFileName, length * sizeof(WCHAR));
```

Comment:
length is already multiplied by 2 at the end of the ConvertToUnicode call.
I think what is wrong is the check if length > MAX_PATH which should be length > MAX_PATH * sizeof(WCHAR) - and of course instead of 2 we should also multiply with sizeof(WCHAR) at the end of ConvertToUnicode.

Reply:
The same applies for the cAlternateFileName conversion too

Reply:
@mfleisz Yes, you're correct. Fixed that.